sdk/go/pulumi/rpc.go

```diff
@@ -184,11 +184,11 @@ func marshalInputAndDetermineSecret(v interface{},
 	destType reflect.Type,
```

Comment:
I assume we could benefit from some deeper testing of this `marshalInputAndDetermineSecret` function?